presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -75,12 +106,14 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
                 Object object = constant.getValue();
                 constantValuesBuilder.add(object);
```

Comment:
`constantValues` is useful for `LARGE_LIST` and `SMALL_LIST_SMALL_PRIMITIVES` only;
`hashBuckets` is useful for `SMALL_LIST` only.
Only generate the ones needed. And set the other one to null.